datalake/src/main/java/com/sequenceiq/datalake/service/upgrade/OrderedOSUpgradeRequestProvider.java

```diff
@@ -0,0 +1,88 @@
+package com.sequenceiq.datalake.service.upgrade;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.osupgrade.OrderedOSUpgradeSet;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.osupgrade.OrderedOSUpgradeSetRequest;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.instancegroup.instancemetadata.InstanceMetaDataV4Response;
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+
+@Component
+public class OrderedOSUpgradeRequestProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrderedOSUpgradeRequestProvider.class);
+
+    private static final String MASTER_GROUP = "master";
+
+    private static final String CORE_GROUP = "core";
+
+    private static final String AUXILIARY_GROUP = "auxiliary";
+
+    private static final String IDBROKER_GROUP = "idbroker";
+
+    private static final String GATEWAY_GROUP = "gateway";
+
+    public OrderedOSUpgradeSetRequest createOrderedOSUpgradeSetRequest(StackV4Response stackV4Response) {
```

Comment:
One minor thing: maybe we can rename it to something like createMediumDutyOrderedOSUpgradeSetRequest?